### ISSUE TITLE
2.0-beta01 crash: IllegalArgumentException: baseUrl must end in /

### ISSUE THREAD

**Issue by eht16** [readrops/Readrops#189]
When starting 2.0-beta01 installed from F-Droid on an existing 1.3 profile, Readrops crashes for me right at startup.

Via logcat, I could see the error message:
```
08-28 23:24:54.459 31500 31554 E [Koin]  : * Instance creation error : could not create instance for '[Factory:'com.readrops.api.services.fever.FeverService']': java.lang.IllegalArgumentException: baseUrl must end in /: https://sub.example.com/freshrss
```

Maybe I need to add a trailing slash to the URL but since the app is crashing at startup I cannot.

<details><summary>Logcat output</summary>
<p>
08-28 23:24:54.459 31500 31554 E [Koin]  : * Instance creation error : could not create instance for '[Factory:'com.readrops.api.services.fever.FeverService']': java.lang.IllegalArgumentException: baseUrl must end in /: https://sub.example.com/freshrss
08-28 23:24:54.459 31500 31554 E [Koin]  : 	retrofit2.Retrofit$Builder.baseUrl(SourceFile:53)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	com.readrops.db.DbModuleKt$$ExternalSyntheticLambda0.invoke(SourceFile:1448)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	org.koin.core.instance.InstanceFactory.create(SourceFile:48)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	org.koin.core.instance.FactoryInstanceFactory.get(SourceFile:1)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.resolveInstance(SourceFile:109)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.get(SourceFile:127)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	com.readrops.db.DbModuleKt$$ExternalSyntheticLambda0.invoke(SourceFile:1536)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	org.koin.core.instance.InstanceFactory.create(SourceFile:48)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	org.koin.core.instance.FactoryInstanceFactory.get(SourceFile:1)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.resolveInstance(SourceFile:109)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.get(SourceFile:127)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	com.readrops.db.DbModuleKt$$ExternalSyntheticLambda0.invoke(SourceFile:752)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	org.koin.core.instance.InstanceFactory.create(SourceFile:48)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	org.koin.core.instance.FactoryInstanceFactory.get(SourceFile:1)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.resolveInstance(SourceFile:109)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.get(SourceFile:127)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	androidx.paging.ChannelFlowCollector.emit(SourceFile:38)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.DistinctFlowImpl$collect$2.emit(SourceFile:763)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.internal.SafeCollectorKt$emitFun$1.invoke(SourceFile:5)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.internal.SafeCollector.emit(SourceFile:10)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.internal.SafeCollector.emit(SourceFile:1)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.FlowKt.emitAllImpl$FlowKt__ChannelsKt(SourceFile:130)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.FlowKt__ChannelsKt$emitAllImpl$1.invokeSuspend(SourceFile:12)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Unknown Source:8)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	kotlinx.coroutines.DispatchedTask.run(Unknown Source:96)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	androidx.work.Worker$2.run(SourceFile:32)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	kotlinx.coroutines.scheduling.TaskImpl.run(Unknown Source:2)
08-28 23:24:54.459 31500 31554 E [Koin]  : 	kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(SourceFile:96)
08-28 23:24:54.460 31500 31554 E [Koin]  : * Instance creation error : could not create instance for '[Factory:'com.readrops.api.services.fever.FeverDataSource']': org.koin.core.error.InstanceCreationException: Could not create instance for '[Factory:'com.readrops.api.services.fever.FeverService']'
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.instance.InstanceFactory.create(SourceFile:185)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.instance.FactoryInstanceFactory.get(SourceFile:1)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.resolveInstance(SourceFile:109)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.get(SourceFile:127)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	com.readrops.db.DbModuleKt$$ExternalSyntheticLambda0.invoke(SourceFile:1536)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.instance.InstanceFactory.create(SourceFile:48)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.instance.FactoryInstanceFactory.get(SourceFile:1)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.resolveInstance(SourceFile:109)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.get(SourceFile:127)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	com.readrops.db.DbModuleKt$$ExternalSyntheticLambda0.invoke(SourceFile:752)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.instance.InstanceFactory.create(SourceFile:48)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.instance.FactoryInstanceFactory.get(SourceFile:1)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.resolveInstance(SourceFile:109)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.get(SourceFile:127)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	androidx.paging.ChannelFlowCollector.emit(SourceFile:38)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.DistinctFlowImpl$collect$2.emit(SourceFile:763)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.internal.SafeCollectorKt$emitFun$1.invoke(SourceFile:5)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.internal.SafeCollector.emit(SourceFile:10)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.internal.SafeCollector.emit(SourceFile:1)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.FlowKt.emitAllImpl$FlowKt__ChannelsKt(SourceFile:130)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.FlowKt__ChannelsKt$emitAllImpl$1.invokeSuspend(SourceFile:12)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Unknown Source:8)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.DispatchedTask.run(Unknown Source:96)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	androidx.work.Worker$2.run(SourceFile:32)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.scheduling.TaskImpl.run(Unknown Source:2)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(SourceFile:96)
08-28 23:24:54.460 31500 31554 E [Koin]  : * Instance creation error : could not create instance for '[Factory:'com.readrops.app.repositories.BaseRepository']': org.koin.core.error.InstanceCreationException: Could not create instance for '[Factory:'com.readrops.api.services.fever.FeverDataSource']'
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.instance.InstanceFactory.create(SourceFile:185)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.instance.FactoryInstanceFactory.get(SourceFile:1)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.resolveInstance(SourceFile:109)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.get(SourceFile:127)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	com.readrops.db.DbModuleKt$$ExternalSyntheticLambda0.invoke(SourceFile:752)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.instance.InstanceFactory.create(SourceFile:48)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.instance.FactoryInstanceFactory.get(SourceFile:1)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.resolveInstance(SourceFile:109)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	org.koin.core.scope.Scope.get(SourceFile:127)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	androidx.paging.ChannelFlowCollector.emit(SourceFile:38)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.DistinctFlowImpl$collect$2.emit(SourceFile:763)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.internal.SafeCollectorKt$emitFun$1.invoke(SourceFile:5)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.internal.SafeCollector.emit(SourceFile:10)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.internal.SafeCollector.emit(SourceFile:1)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.FlowKt.emitAllImpl$FlowKt__ChannelsKt(SourceFile:130)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.flow.FlowKt__ChannelsKt$emitAllImpl$1.invokeSuspend(SourceFile:12)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Unknown Source:8)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.DispatchedTask.run(Unknown Source:96)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	androidx.work.Worker$2.run(SourceFile:32)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.scheduling.TaskImpl.run(Unknown Source:2)
08-28 23:24:54.460 31500 31554 E [Koin]  : 	kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(SourceFile:96)
08-28 23:24:54.481  1654  4256 I ActivityTaskManager: START u0 {flg=0x14008000 cmp=com.readrops.app/.util.CrashActivity (has extras)} from uid 10222
08-28 23:24:54.483  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 265464455; UID 10222; state: DISABLED
08-28 23:24:54.483  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 265452344; UID 10222; state: DISABLED
08-28 23:24:54.484  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 266124927; UID 10222; state: DISABLED
08-28 23:24:54.485  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 265451093; UID 10222; state: DISABLED
08-28 23:24:54.485  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 265456536; UID 10222; state: DISABLED
08-28 23:24:54.485  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 255940284; UID 10222; state: DISABLED
08-28 23:24:54.486  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 236283604; UID 10222; state: DISABLED
08-28 23:24:54.486  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 263959004; UID 10222; state: DISABLED
08-28 23:24:54.486  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 264304459; UID 10222; state: DISABLED
08-28 23:24:54.486  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 264301586; UID 10222; state: DISABLED
08-28 23:24:54.487  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 254631730; UID 10222; state: DISABLED
08-28 23:24:54.487  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 273509367; UID 10222; state: DISABLED
08-28 23:24:54.487  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 263259275; UID 10222; state: DISABLED
08-28 23:24:54.489  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 194480991; UID 10222; state: ENABLED
08-28 23:24:54.490  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 174042980; UID 10222; state: DISABLED
08-28 23:24:54.493  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 184838306; UID 10222; state: DISABLED
08-28 23:24:54.493  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 185004937; UID 10222; state: DISABLED
08-28 23:24:54.494  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 181136395; UID 10222; state: DISABLED
08-28 23:24:54.494  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 174042936; UID 10222; state: DISABLED
08-28 23:24:54.498  1654  4256 D CompatibilityChangeReporter: Compat change id reported: 205907456; UID 10222; state: ENABLED
08-28 23:24:54.502  1654  4256 I ActivityTaskManager: Add Task{9a91f71 #21128 type=standard A=10222:com.readrops.app} to hidden list because adding Task{1558f24 #21129 type=standard A=10222:com.readrops.app}
08-28 23:24:54.505  2676  2819 W Parcel  : Expecting binder but got null!
08-28 23:24:54.507  1654  1596 D CompatibilityChangeReporter: Compat change id reported: 168419799; UID 10222; state: DISABLED
08-28 23:24:54.507  1654  1596 D CompatibilityChangeReporter: Compat change id reported: 273564678; UID 10222; state: DISABLED
08-28 23:24:54.507 31500 31554 I com.readrops.app: System.exit called, status: 0
08-28 23:24:54.510  1654  1596 D CoreBackPreview: Window{1ee7990 u0 Splash Screen com.readrops.app}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@ec928e, mPriority=0}
08-28 23:24:54.552  1654 18440 I WindowManager: WIN DEATH: Window{d80f01d u0 com.readrops.app/com.readrops.app.MainActivity}
08-28 23:24:54.553  1654 18440 W InputManager-JNI: Input channel object 'd80f01d com.readrops.app/com.readrops.app.MainActivity (client)' was disposed without first being removed with the input manager!
08-28 23:24:54.553  1654  1596 I ActivityManager: Process com.readrops.app (pid 31500) has died: fg  TOP 
08-28 23:24:54.559  1654  1596 W ActivityTaskManager: Force removing ActivityRecord{ea5f5d7 u0 com.readrops.app/.MainActivity} t21128}: app died, no saved state
08-28 23:24:54.566  1654  1845 D CompatibilityChangeReporter: Compat change id reported: 135634846; UID 10222; state: DISABLED
08-28 23:24:54.566  1654  1845 D CompatibilityChangeReporter: Compat change id reported: 177438394; UID 10222; state: DISABLED
08-28 23:24:54.566  1654  1845 D CompatibilityChangeReporter: Compat change id reported: 135772972; UID 10222; state: DISABLED
08-28 23:24:54.566  1654  1845 D CompatibilityChangeReporter: Compat change id reported: 135754954; UID 10222; state: ENABLED
08-28 23:24:54.566  1654  1862 D CompatibilityChangeReporter: Compat change id reported: 143937733; UID 10222; state: ENABLED
08-28 23:24:54.567  1654 18440 D CoreBackPreview: Window{96a1873 u0 Splash Screen com.readrops.app EXITING}: Setting back callback null
08-28 23:24:54.567  1654  1596 W InputManager-JNI: Input channel object '96a1873 Splash Screen com.readrops.app (client)' was disposed without first being removed with the input manager!
08-28 23:24:54.668 31566 31566 W com.readrops.app: ART APEX data files are untrusted.
08-28 23:24:54.770 31566 31566 D com.readrops.app: Time zone APEX ICU file found: /apex/com.android.tzdata/etc/icu/icu_tzdata.dat
08-28 23:24:54.770 31566 31566 D com.readrops.app: I18n APEX ICU file found: /apex/com.android.i18n/etc/icu/icudt70l.dat
</p>
</details>

**Comment by Shinokuni** [comment]
What kind of account are you using? From the stacktrace, it seems that Readrops try to instantiate Fever dependencies, but as you might use an other account type, there would be a conflict between url rules.

**Comment by Shinokuni** [comment]
I think your probleme should be solved in v2.0-beta02. I submitted it a few days ago to F-Droid, could you try out when it's available?

**Comment by eht16** [comment]
> What kind of account are you using? From the stacktrace, it seems that Readrops try to instantiate Fever dependencies, but as you might use an other account type, there would be a conflict between url rules.

It is a FreshRSS account, it worked fine before the update. I did not change any settings.

> I think your probleme should be solved in v2.0-beta02. I submitted it a few days ago to F-Droid, could you try out when it's available?

Just tried and works again, thank you.
